### PR TITLE
change plotfile RZ allowed diag to r,t from x,y

### PR DESCRIPTION
**Pull request by RTSandberg**
Previously there were multiple diagnostic naming conventions in RZ: openPMD RZ transverse field were `r`,`t` and plotfile RZ transverse fields were `x`,`y`.  Furthermore, if `<diag>.dump_rz_modes=1` and all modes are written to plotfile then the modes are labeled as `<field>r_<mode>_<real/imag>`.  Now all RZ transverse fields are labeled `r` and `t`.

### CODE REVIEW

File: Python/pywarpx/picmi.py [hunk 1]
```diff
@@ -1744,27 +1744,39 @@ def initialize_inputs(self):
...
-                    fields_to_plot.add('By')
-                    fields_to_plot.add('Bz')
+                    for field_name in B_fields_list:
+                        fields_to_plot.add(field_name)
                 elif dataname == 'J':
-                    fields_to_plot.add('jx')
-                    fields_to_plot.add('jy')
-                    fields_to_plot.add('jz')
+                    for field_name in J_fields_list:
+                        fields_to_plot.add(field_name.lower())
```
Comment (EZoni) [inline Python/pywarpx/picmi.py:1767]: What does `lower()` do here and why is it done only for `J`?

File: Source/Diagnostics/FullDiagnostics.cpp [hunk 2]
```diff
@@ -635,7 +619,49 @@ FullDiagnostics::InitializeFieldFunctors (int lev)
...
+            } else if ( m_varnames[comp] == "Bt" ){
+                m_all_field_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_Bfield_aux(lev, 1), lev, m_crse_ratio);
+            } else if ( m_varnames[comp] == "jr" ){
+                m_all_field_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_current_fp(lev, 0), lev, m_crse_ratio);
+            } else if ( m_varnames[comp] == "jt" ){
+                m_all_field_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_current_fp(lev, 1), lev, m_crse_ratio);
+            } else if ( m_varnames[comp] == "Ar" ){
+                m_all_field_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_vector_potential_fp(lev, 0), lev, m_crse_ratio);
+            } else if ( m_varnames[comp] == "At" ){
+                m_all_field_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_vector_potential_fp(lev, 1), lev, m_crse_ratio);
```
Comment (EZoni) [inline Source/Diagnostics/FullDiagnostics.cpp:639]: I'm noticing that this seems to be done for the `fp` data only. Is the `cp` completely missing from the file?

**Review by EZoni — COMMENTED** [review]
Thank you for this PR, @RTSandberg! It looks good to me, I just left a couple of comments.

**Comment by EZoni** [comment]
@dpgrote Dave, I feel it could be useful if you chime in here and check that the changes look good to you.

File: Python/pywarpx/picmi.py (hunk 1 above)
Comment (RTSandberg) [reply Python/pywarpx/picmi.py:1767]: ``lower()`` changes the case of the string to all lower case.  This was done because that is how ``J`` is treated elsewhere in the input parsing.  Also, ``j`` is given in lower case in the documentation.

**Comment by RTSandberg** [comment]
I just realized that I only made this change to the Full Diagnostic, not the Back-Transformed diagnostic.

File: Source/Diagnostics/FullDiagnostics.cpp (hunk 2 above)
Comment (EZoni) [reply Source/Diagnostics/FullDiagnostics.cpp:639]: @RevathiJambunathan  
It seems like we don't do the same in this file for the `_cp` data/pointers (for example for the current J) inside `InitializeFieldFunctors`: does it mean that the coarse-patch field functors are initialized in some other file? I'm asking to know if Ryan should update the RZ part of those, too, somewhere else.

File: Docs/source/usage/parameters.rst [hunk 3]
```diff
@@ -2148,12 +2148,11 @@ In-situ capabilities can be used by turning on Sensei or Ascent (provided they a
     Possible scalar fields: ``part_per_cell`` ``rho`` ``phi`` ``F`` ``part_per_grid`` ``divE`` ``divB`` and ``rho_<species_name>``, where ``<species_name>`` must match the name of one of the available particle species. Note that ``phi`` will only be written out when do_electrostatic==labframe. Also, note that for ``<diag_name>.diag_type = BackTransformed``, the only scalar field currently supported is ``rho``.
     Possible vector field components in Cartesian geometry: ``Ex`` ``Ey`` ``Ez`` ``Bx`` ``By`` ``Bz`` ``jx`` ``jy`` ``jz``.
     Possible vector field components in RZ geometry: ``Er`` ``Et`` ``Ez`` ``Br`` ``Bt`` ``Bz`` ``jr`` ``jt`` ``jz``.
-    Default is ``<diag_name>.fields_to_plot = Ex Ey Ez Bx By Bz jx jy jz``,
-    unless in RZ geometry with ``<diag_name>.format == openpmd``,
-    then default is ``<diag_name>.fields_to_plot = Er Et Ez Br Bt Bz jr jt jz``.
+    The default is ``<diag_name>.fields_to_plot = Ex Ey Ez Bx By Bz jx jy jz``,
+    unless in RZ geometry,
+    then the default is ``<diag_name>.fields_to_plot = Er Et Ez Br Bt Bz jr jt jz``.
```
Comment (EZoni) [inline Docs/source/usage/parameters.rst]: This could be maybe rephrased (more clearly?) as
```suggestion
    The default is ``<diag_name>.fields_to_plot = Ex Ey Ez Bx By Bz jx jy jz`` in Cartesian geometry and ``<diag_name>.fields_to_plot = Er Et Ez Br Bt Bz jr jt jz`` in RZ geometry.
```

File: Examples/Tests/silver_mueller/analysis_silver_mueller_rz.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+# Copyright 2021
+#
+# This file is part of WarpX.
+#
+# License: BSD-3-Clause-LBNL
+"""
+This tests verifies the Silver-Mueller boundary condition.
+A laser pulse is emitted and propagates towards the boundaries ; the
+test check that the reflected field at the boundary is negligible.
```
Comment (EZoni) [inline Examples/Tests/silver_mueller/analysis_silver_mueller_rz.py]:
```suggestion
This test verifies the Silver-Mueller boundary condition.
A laser pulse is emitted and propagates towards the boundaries;
the test checks that the reflected field at the boundary is negligible.
```

File: Examples/Tests/silver_mueller/analysis_silver_mueller_rz.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+all_data_level_0 = ds.covering_grid(level=0,left_edge=ds.domain_left_edge, dims=ds.domain_dimensions)
+Er = all_data_level_0['boxlib', 'Er'].v.squeeze()
+Et = all_data_level_0['boxlib', 'Et'].v.squeeze()
+Ez = all_data_level_0['boxlib', 'Ez'].v.squeeze()
+# The peak of the initial laser pulse is on the order of 6 V/m
+# Check that the amplitude after reflection is less than 0.01 V/m
+max_reflection_amplitude = 0.01
+assert np.all( abs(Er) < max_reflection_amplitude )
+assert np.all( abs(Et) < max_reflection_amplitude )
+assert np.all( abs(Ez) < max_reflection_amplitude )
```
Comment (EZoni) [inline Examples/Tests/silver_mueller/analysis_silver_mueller_rz.py]: Is this analysis here significantly different from what was done in `analysis_silver_mueller.py` (the original Cartesian version)? If not, we could simply have an `if` condition in the original analysis script, to distinguish the Cartesian and RZ cases. And we could parse the output file `warpx_used_inputs` to learn which test is being run (this is already done in other analysis scripts). If the differences are minimal, I would be in favor of not duplicating the analysis script. Especially because we don't do it consistently throughout the WarpX test suite (we don't enforce the policy "one test - one input - one analysis script", it was discussed in the past, but never done).

**Review by EZoni — COMMENTED** [review]
Thank you, @RTSandberg! I left some more comments in the review.

Also, since you now made the change for BTD, too, on the PICMI side, shouldn't we make changes similar to those in FullDiagnostics.cpp also in BTDiagnostics.cpp (for the function `BTDiagnostics::InitializeFieldFunctors`), to complete the BTD part?

**Review by RTSandberg — COMMENTED** [review]
Clarifying suggestions were accepted

File: Examples/Tests/silver_mueller/analysis_silver_mueller_rz.py (hunk 5 above)
Comment (RTSandberg) [reply Examples/Tests/silver_mueller/analysis_silver_mueller_rz.py]: The analysis is not significantly different, this is a good idea.  I did not think to use ``warpx_used_inputs`` to do logic on the geometry. Thanks, @EZoni !

**Comment by EZoni** [comment]
> Also, since you now made the change for BTD, too, on the PICMI side, shouldn't we make changes similar to those in FullDiagnostics.cpp also in BTDiagnostics.cpp (for the function BTDiagnostics::InitializeFieldFunctors), to complete the BTD part?

Clarified offline, this is done separately in the function `InitializeFieldFunctorsRZopenPMD` for RZ geometry.

File: Examples/Tests/silver_mueller/analysis_silver_mueller.py [hunk 6]
```diff
@@ -24,15 +25,27 @@
 
 ds = yt.load( filename )
 all_data_level_0 = ds.covering_grid(level=0,left_edge=ds.domain_left_edge, dims=ds.domain_dimensions)
-Ex = all_data_level_0['boxlib', 'Ex'].v.squeeze()
-Ey = all_data_level_0['boxlib', 'Ey'].v.squeeze()
-Ez = all_data_level_0['boxlib', 'Ez'].v.squeeze()
+warpx_used_inputs = open('./warpx_used_inputs', 'r').read()
+if re.search('geometry.dims\s*=\s*RZ', warpx_used_inputs):
```
Comment (EZoni) [inline Examples/Tests/silver_mueller/analysis_silver_mueller.py]: Just a suggestion, maybe we could have an auxiliary boolean and use `re.search` only once:
```suggestion
geom_RZ = re.search('geometry.dims = RZ', warpx_used_inputs)
if geom_RZ:
```
Note that I also removed the `\s*` expression, the output in that file should always be written with one single white space, so this should work.

File: Examples/Tests/silver_mueller/analysis_silver_mueller.py [hunk 7]
```diff
@@ -24,15 +25,27 @@
...
+    Ey = all_data_level_0['boxlib', 'Ey'].v.squeeze()
+    Ez = all_data_level_0['boxlib', 'Ez'].v.squeeze()
 # The peak of the initial laser pulse is on the order of 6 V/m
 # Check that the amplitude after reflection is less than 0.01 V/m
 max_reflection_amplitude = 0.01
-assert np.all( abs(Ex) < max_reflection_amplitude )
-assert np.all( abs(Ey) < max_reflection_amplitude )
-assert np.all( abs(Ez) < max_reflection_amplitude )
+
+if re.search('geometry.dims\s*=\s*RZ', warpx_used_inputs):
```
Comment (EZoni) [inline Examples/Tests/silver_mueller/analysis_silver_mueller.py]: And we could reuse the boolean here:
```suggestion
if geom_RZ:
```

**Review by EZoni — COMMENTED** [review]
Thank you for the latest changes, @RTSandberg! I added only one small suggestion.

File: Examples/Tests/silver_mueller/analysis_silver_mueller.py (hunk 6 above)
Comment (RTSandberg) [reply Examples/Tests/silver_mueller/analysis_silver_mueller.py]: ok that sounds good to me.  Making the change now

**Review by EZoni — APPROVED** [review]
